server/utils/terminal/terminal_run.py

```
@@ -203,3 +213,34 @@ def terminal_run(
+        sig = check_signature(cmd)
+        if sig.matched:
+            logger.warning("[Guardrails:L2] BLOCKED cmd=%s technique=%s rule=%s", cmd[:100], sig.technique, sig.rule_id)
+            return CompletedProcess(args=args, returncode=126, stdout="", stderr=f"Blocked by safety guardrail: {sig.description}")
+
+    if config.llm_judge:
+        uid, sid = None, None
+        try:
+            ctx = get_user_context()
+            uid, sid = ctx.get("user_id"), ctx.get("session_id")
+        except Exception:
+            pass
```

Ruff (0.15.10)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)